### PR TITLE
[C-6478] Audit Events

### PR DESCRIPTION
**Pull request by tk26**
## Description of the change

Audit events 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
Screen Shot 2022-07-25 at 7 21 44 PM

### Code review 
- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9, 3.10.4]
```
Comment (tk26) [inline .github/workflows/pythonpackage.yml:20]: GH actions didn't like 3.10

File: tests/test_audit_events.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      "actor": {
+        "email": "foo@bar.com",
+        "id": "foo-user"
+      },
+      "auditEventId": "ZX3xXUMNKL4y2NkiKgstl",
+      "source": "courier.studio",
+      "target": {},
+      "timestamp": "2022-07-22T22:33:59.552Z",
+      "type": "notification:published",
+      "workspaceId": "my-workspace"
```
Comment (scarney81) [inline tests/test_audit_events.py:33]: We should not be exposing workspaceId publicly

File: trycourier/audit_events.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            raise CourierAPIException(resp)
+
+        return resp.json()
+
+    def list_audit_events(self, cursor=None):
+        """
+        Lists audit events
+
+        Args:
+            cursor: optional pointer to for fetching subsequent result pages
```
Comment (StephenYin01) [inline trycourier/audit_events.py:41]: optional pointer to where?

**Review by StephenYin01 — COMMENTED** [review]
just one question on what I think is a typo in the comments, other than that LGTM

File: trycourier/audit_events.py (hunk 3 above)
Comment (tk26) [inline trycourier/audit_events.py:41]:
```suggestion
            cursor: optional pointer to fetch subsequent result pages
```

**Review by StephenYin01 — APPROVED** [review]

**Pull request merged**